src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -390,6 +390,59 @@ public object VisitTable(TableExpression tableExpression, SearchOptions context)
                         _includeCtes = new List<string>();
                     }
 
+                    _includeCtes.Add(TableExpressionName(_tableExpressionCounter));
+                    break;
+                case TableExpressionKind.RevInclude:
+                    var truncatedItemsLimit = 100;
+                    var revIncludeExpression = (RevIncludeExpression)tableExpression.NormalizedPredicate;
+
+                    StringBuilder.Append("SELECT DISTINCT TOP " + (truncatedItemsLimit + 1) + " ");
+                    StringBuilder.Append(VLatest.Resource.ResourceSurrogateId, referenceSourceTableAlias).AppendLine(" AS Sid1, 0 AS IsMatch")
```

Comment:
Question: If I have 	http://hapi.fhir.org/baseR4/Medication?_revinclude=MedicationRequest:medication&_count=5 for example, it will first do _count=5 and then  INNER JOIN due to the expression order - right?

Reply:
(notice that this comment refers to my new revision).
correct. if you do _count=5 it will add that TOP 5 in cte1
if for example you will do _id=someguid it will add it (in that case) in cte0.

cte0 and cte1 examples are available in the snippet I paste in this PR description (or see the generated sql)